### DISCUSSION TITLE
(3) session not authenticated despite PUBLIC write/read access settings and PUBLIC screens statuses

### DISCUSSION THREAD

**Discussion by vlakula** [Budibase/budibase#14655] [Help] [answered]
Hi!
I have tried to open my app to public and there is a problem with "(3) session not authenticated" error.
First of all I have read [this thread](https://github.com/Budibase/budibase/issues/4582) and [this thread](https://github.com/Budibase/budibase/discussions/6012) and turned on PUBLIC setting in database and PUBLIC setting for each screen of my app. After that I have publish app again and the "(3) session not authenticated" error repeats inspite of all actions above.

Here are the settings of database and screens:
![image](https://github.com/user-attachments/assets/df857d97-2eff-4cb5-98da-c4430b96e232)
![image](https://github.com/user-attachments/assets/03828d56-0c80-419c-9bee-01352e841dea)
![image](https://github.com/user-attachments/assets/e2f899dc-46f6-47e8-b390-584b236b1672)
Here are an errors if I try to open app via some device where I didn't login to Budibase account:
![image](https://github.com/user-attachments/assets/5f6c2e15-67ae-443e-8e9a-874afec13753)
![image](https://github.com/user-attachments/assets/7e677df2-a958-450d-bcd7-140a345f4973)

So at the moment I can't start to work with my users. Could you help me please?
Thank you!

**Comment by ConorWebb96** [comment]
Hey @vlakula,

Do you have any queries on your screen? It's outputting three times, which suggests that there are three different areas on your screen for which you don't have public permissions set.

Can you provide a screenshot of your component tree?

**Reply by vlakula** [reply]
@ConorWebb96 thanks for so fast reply!
Of course, here is the component tree of the main screen (for example, but all of my screens should be public):
![image](https://github.com/user-attachments/assets/6c42f9f3-c175-4321-ba86-c045bbcf701b)
If there will be helpful here are the settings of Data Provider and Repeater on the main screen:
![image](https://github.com/user-attachments/assets/bfc99b5f-a4d5-42b8-a772-24025232dc56)
![image](https://github.com/user-attachments/assets/e2e1b5ae-d755-413f-9e4f-ab2a5f272327)

**Reply by ConorWebb96** [reply]
Hey @vlakula,

I was hoping you'd have more than one data provider on the screen. This isn't the case. If possible, can you provide me with an app export so I can have a more in-depth look? 👀 Feel free to export the app without any data.

**Reply by vlakula** [reply]
@ConorWebb96 here it is!
Hope I made it correct (because I exported project in Budibase the first time):
[Craftkino Movie Library-export-1727686010942.tar.gz](https://github.com/user-attachments/files/17186885/Craftkino.Movie.Library-export-1727686010942.tar.gz)

**Reply by ConorWebb96** [reply, marked as answer]
Hey @vlakula,

You need to change the access on the specific table you wish to make publicly accessible. I've included a screenshot of this below.

<img width="1058" alt="Screenshot 2024-09-30 at 10 19 10" src="https://github.com/user-attachments/assets/e3eb0997-d41d-43fb-86fc-6856a48df13e">

You can do this by selecting the table and then clicking `Access`  at the top. I've highlighted this below.
<img width="1073" alt="Screenshot 2024-09-30 at 10 26 01" src="https://github.com/user-attachments/assets/0fa00b5c-7847-472b-8087-b413b7bc63d5">

**Reply by vlakula** [reply]
@ConorWebb96 it's really works!
I am very grateful to you. And special thanks for such a prompt response.
Thank you!
